### PR TITLE
Humoud AlGhanim

### PR DESCRIPTION
**Pull request by HAlGhanim**

### CODE REVIEW

File: src/objects.ts [hunk 1]
```diff
@@ -27,8 +27,12 @@ function createBook(
...
 ): Book {
   // write your code here...
-
-  return {} as Book; // replace "{} as Book" with what you see is fit
+  return {
+    title: title,
+    author: author,
+    publishedYear: publishedYear,
+    genre: genre,
+  }; // replace "{} as Book" with what you see is fit
```
Comment (Octowl) [inline src/objects.ts:35]: A fun thing you can do in JS: if the key and the variable share the same name you can do:
```ts
return {
  title,
  author,
  publishedYear,
  genre
};
```

**Review by Octowl — COMMENTED** [review]
This PR made me realise I'm missing an edge condition for this task. I've now made it slightly harder and doubled the size of the test suite 😬

File: src/challenge.ts [hunk 2]
```diff
@@ -39,7 +39,9 @@ function addReview(
   comment: string
 ): ReviewedBook {
   // write your code here...
-
+  book.reviews
+    ? book.reviews.push({ reviewer: reviewer, comment: comment })
+    : (book.reviews = [{ reviewer: reviewer, comment: comment }]);
```
Comment (Octowl) [inline src/challenge.ts:44]: Ternary operators should only really be used as _expressions_ (where the entire ternary operator will be replaced with a value. They should not have side-effects or be used as a general replacement for if statements.

A better ternary use here would be:
```TS
  const newReview: Review = { reviewer, comment };

  book.reviews =
    book.reviews ? [...book.reviews, newReview] : [newReview];
 ```
